source/common/http/http2/metadata_encoder.cc

```diff
@@ -142,6 +142,9 @@ class BufferMetadataSource : public http2::adapter::MetadataSource {
     return std::make_pair(static_cast<int64_t>(to_copy), payload_.length() == 0);
   }
 
+  // TODO: implement this method
+  void OnFailure() override {}
```

Comment:
Do you have any idea what implementing this method might entail?

Reply:
I'm not quite sure how this method should be implemented here.
The other subclass is SelfDeletingMetadataSource but I think these 2 classes are quite different.
Adding @birenroy @diannahu who might know more about this.
We could wait for an actual impl to merge this PR or we can land first and then implement later. Either way I'm happy.

Reply:
Makes sense. I suspect that the current no-op implementation is safe to land but it does make me wonder. How 'bout we wait for either of @birenroy or @diannahu to comment and then land based on their input?

Reply:
Hi, thanks for adding us! I added this interface method for SelfDeletingMetadataSource to know to delete itself on failure. AIUI, SelfDeletingMetadataSource is only used internally to NgHttp2Adapter, so it should be okay for other implementations (including BufferMetadataSource) to leave OnFailure() as a no-op. Thanks!

Reply:
Perfect Thanks, Dianna! In that case, @RenjieTang perhaps we should remove the TODO, then we can land it.